Sources/Commands/GenerateShellScript.swift

```diff
@@ -0,0 +1,328 @@
+    _ = root.add(option: "-help", kind: Bool.self, usage: "Display available options")
+    _ = root.add(option: "-I", kind: String.self, usage: "Add directory to the import search path")
+    _ = root.add(option: "-j", kind: Int.self, usage: "Number of commands to execute in parallel")
+    _ = root.add(option: "-L", kind: String.self, usage: "Add directory to library link search path")
+//    -l<value>              Specifies a library which should be linked against
+    _ = root.add(option: "-module-cache-path", kind: String.self, usage: "Specifies the Clang module cache path")
+    _ = root.add(option: "-module-link-name", kind: String.self, usage: "Library to link against when using this module")
+    _ = root.add(option: "-module-name", kind: String.self, usage: "Name of the module to build")
+    _ = root.add(option: "-nostdimport", kind: Bool.self, usage: "Don't search the standard library import path for modules")
+    _ = root.add(option: "-num-threads", kind: Int.self, usage: "Enable multi-threading and specify number of threads")
+    _ = root.add(option: "-Onone", kind: Bool.self, usage: "Compile without any optimization")
+    _ = root.add(option: "-Ounchecked", kind: Bool.self, usage: "Compile with optimizations and remove runtime safety checks")
+    _ = root.add(option: "-O", kind: Bool.self, usage: "Compile with optimizations")
+    _ = root.add(option: "-sdk", kind: String.self, usage: "Compile against <sdk>")
+    _ = root.add(option: "-static-executable", kind: Bool.self, usage: "Statically link the executable")
+    _ = root.add(option: "-static-stdlib", kind: Bool.self, usage: "Statically link the Swift standard library")
+    _ = root.add(option: "-suppress-warnings", kind: Bool.self, usage: "Suppress all warnings")
+    _ = root.add(option: "-swift-version", kind: String.self, usage: "Interpret input according to a specific Swift language version number")
+    _ = root.add(option: "-target-cpu", kind: String.self, usage: "Generate code for a particular CPU variant")
+    _ = root.add(option: "-target", kind: String.self, usage: "Generate code for the given target")
+//    -use-ld=<value>        Specifies the linker to be used
+    _ = root.add(option: "-version", kind: Bool.self, usage: "Print version information and exit")
+    _ = root.add(option: "-v", kind: Bool.self, usage: "Show commands to run and use verbose output")
+    _ = root.add(option: "-warnings-as-errors", kind: Bool.self, usage: "Treat warnings as errors")
+    _ = root.add(option: "-Xcc", kind: String.self, usage: "Pass <arg> to the C/C++/Objective-C compiler")
+    _ = root.add(option: "-Xlinker", kind: String.self, usage: "Specifies an option which should be passed to the linker")
+
+    switch shell {
+    case .bash: bash(root: root, print: print)
+    case .zsh: zsh(root: root, print: print)
+    }
+}
+
+
+// MARK:- BASH
```

Comment:
That is a great point. However at the moment ArgumentParser isn't sufficient to represent the arguments of the `swift` binary; i.e. both the 3 tools and the compiler. So while I can agree with you it would be nice to generate scripts for any ArgumentParser, we need a solution for the "root" argument parser.